src/CoreWCF.Primitives/src/CoreWCF/Channels/BinaryMessageEncoderFactory.cs

```diff
@@ -568,21 +566,22 @@ private ArraySegment<byte> ExtractSessionInformationFromMessage(ArraySegment<byt
                     }
                 }
 
-                return new ArraySegment<byte>(buffer, newOffset, newSize);
+                return messageData.Slice(headerSize + dictionarySize, messageData
+                    .Length - headerSize - dictionarySize);
```

Comment:
I think you can skip the second parameter here as it will default to the rest of the sequence after the offset.

Reply:
done https://github.com/CoreWCF/CoreWCF/pull/801/commits/652e27f4d056acbe8836bfa2eb4ebb7635c0dd76